Methods2/Program.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+
+namespace Methods2
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            //2
+            int debt = 700;
+            Console.WriteLine("Your current debt is {0:#}", debt);
+            Console.WriteLine("Enter sum to pay off debt: ");
+            int money = Int32.Parse(Console.ReadLine());
+
+            Amortization(money, ref debt);
+
+            //3
+            //recursive calls are slower than iterative calls.
+            //reusable method calls may overflow the system stack
+            int N = 6;
+            int resultRec = FactorialRec(N);
+            Console.WriteLine("Number of options {0:#}", resultRec);
+
+            //4
+            Console.WriteLine("Enter the operand1 and operand2(Integer): ");
+            int operand1 = Int32.Parse(Console.ReadLine());
+            int operand2 = Int32.Parse(Console.ReadLine());
+            long result;
+
+            Console.WriteLine("Enter key for operation:\n" +
+                              "r - remainder of division\n" +
+                              "p - pow\n" +
+                              "c - concatenationpcd\n" +
+                              "d - division\n");
+            string sign = Console.ReadLine();
+
+            switch (sign)
+            {
+                case "r":
+                    result = RemainderOfDivision(operand1, operand2);
+                    Console.WriteLine("{0} % {1} = {2:#}", operand1, operand2, result);
+                    break;
+                case "p":
+                    result = Pow(operand1, operand2);
+                    Console.WriteLine("{0} ^ {1} = {2:G}", operand1, operand2, result);
+                    break;
+                case "c":
+                    result = Concatenation(operand1, operand2);
+                    Console.WriteLine("{0} + {1} = {2:#}", operand1, operand2, result);
+                    break;
+                case "d":
+                    result = Division(operand1, operand2);
+                    Console.WriteLine("{0} / {1} = {2:G}", operand1, operand2, result);
+                    break;
+                default:
+                    Console.WriteLine("You entered another word");
+                    break;
+            }
+
+            //5(additional task)
+            double a = 5;
+            double b = 8;
+            double c = 34;
+            Calculate(ref a, ref b, ref c);
```

Comment:
`ref` is not suitable param type here
Think and write why?

Reply:
I think param ref no need here.
Need to add to the function Console.WriteLine(a, b, c);

Reply:
I mean you need use `out` here. But why ?

Reply:
I try use out, but there was a mistake "use of unassigned out parameter"

Reply:
Disregard, leave `ref`